### DISCUSSION TITLE
v1 SLA of E-Fil+ Pilot Notary Expectations

### DISCUSSION THREAD

**Discussion by kevzak** [filecoin-project/notary-governance#807] [E-Fil+ WG]
v0 of the E-Fil+ Pilot started in Nov 2022. Since then, we've had 11 private data applications submitted and reviewed with varying success. [LINK](https://efilplus.super.site/e-fil-client-applicants-list)

Regarding Notary participation: One of the main tests of the pilot was regarding notary accountability. We were looking to assign notaries, randomly, to each application and assess if they reviewed and approved or denied the application versus an expected timeframe.

However, there were issues with random assignment logic in github, also, instructions for notaries to complete a check were not clear regarding required commenting.

Therefore, we want to update the SLA and create v1 of the E-Fil+ Notary expectations for the pilot moving forward as follows:
1) Two notaries will be chosen sequentially from the E-Fil multisig master list and tagged in each subsequent github application review request 
2) Signing expectations:
    a) Notaries should not sign on applications in the E-Fil+ program that they were not assigned to.
3) Timeframe expectations:
    a) Once tagged, notaries are expected to review the application, and at a minimum leave a comment about the application, within 72 hours (standard business hours) turnaround time of being tagged.
4) Expectations for review:
    a) If a tagged notary has a conflict of interest, they can comment "I am unavailable to participate in review of this application" and the next notary will be tagged to take their place.
    b) If a tagged notary has reviewed the application and decided to sign, they should also leave a comment as to why they are signing to confirm they have completed proper due diligence
    c) If a tagged notary has reviewed the application and decided not to sign for any reason unrelated to the application details, they should leave a comment as to why they are not signing. Ex: "I am choosing not to review this application as it is out of my region"
    d) If a tagged notary has reviewed the application and decided not to sign because they felt specific information was missing, they should leave a comment as to why they are not signing

Recommended application categories to comment on include:
- Proper KYC/KYB
- Data use case, proof of data size request
- Storage Plan Details (SPs, miner IDs, location)
- Filecoin use case (new/experienced user, understanding of how to use Filecoin processes)

Overall E-Fil+ Notary Expectations:
- During v1, statistics of E-Fil+ notary participation will be tracked. Notaries who do not meet participation expectations by at least commenting on the application within the SLA timeframe will be noted and after 3 missed opportunities will be removed from further participation in the pilot. 

Related posts:
https://github.com/filecoin-project/notary-governance/issues/611
https://github.com/filecoin-project/notary-governance/discussions/810#discussion-4762993

**Comment by kevzak** [comment]
**PARTICIPATION CONFIRMATION:**
-Any interested notaries please comment below and confirm you have reviewed participation expectations above.
- Current E-Fil+ Notaries, I ask you to review details above and comment below if you are interested in continuing participation.


@jamerduhgamer
@PluskitOfficial
@cryptowizzard
@davidthoms
@xingjitansuo
@swatchliu
@BlockMakeronline
@MatrixStorage
@Fenbushi-Filecoin
@bmcnabb25
@flyworker
@neogeweb3
@newwebgroup
@derricktan23
@liyunzhi-666
@Fatman13
@IPFSUnion
@stcouldlisa 
@MRJAVAZHAO
@1475notary

**Reply by Fatman13** [reply]
Reviewed and would like to continue participation. 👍

**Reply by bmcnabb25** [reply]
Also willing to continue participation!

**Reply by Destore2023** [reply]
Reviewed and willing to keep participating!

**Reply by Bennyyangpu** [reply]
@kevzak we would also like to participate.

**Reply by Wengeding** [reply]
We are happy to participate and provide support @kevzak

**Comment by panges2** [comment]
+1 Excited to see the outcome of this!

**Reply by Carohere** [reply]
Amazing work. Willing to help!

**Reply by TakiChain** [reply]
We're willing to participate in v1.

**Comment by Joss-Hua** [comment]
Glad to see the significant progress of E-FIL after a quarter of polishing.
Express my interest here and be willing to participate, and provide support.

**Reply by kevzak** [reply]
> Glad to see the significant progress of E-FIL after a quarter of polishing. Express my interest here and be willing to participate, and provide support.

@Joss-Hua please send your slack and Notary application, I cannot find you there

**Reply by Joss-Hua** [reply]
My [Slack](https://filecoinproject.slack.com/archives/DNEDPLEDB): Joss-Venus.
Notary application of [V3](https://github.com/filecoin-project/notary-governance/issues/431).
Notary application of [V4](https://github.com/filecoin-project/notary-governance/issues/645).

**Comment by liyunzhi-666** [comment]
After working at E-Fil+ WG for the past few months, I think Kevin-Z @kevzak has done a great job.
This SLA statement I think fits well with the objectives of E-Fil+ WG's work. I would like to continue to work as a Notary for E-Fil+ and help in any way I can.

**Comment by Fenbushi-Filecoin** [comment]
@kevzak Thanks for your time and effort. We are willing to participate in E-Fil and provide support.

**Reply by Casey-PG** [reply]
@kevzak we're willing to participate too. FYI this week is CNY holidays, more notaries will likely be involved if the deadline can be postponed a bit.

**Comment by derricktan23** [comment]
@kevzak  , Kudos to Kevin's patience and professionalism throughout the E-Fil+ Pilot phase.  We are willing to participate in E-Fil and provide support.

**Comment by flyworker** [comment]
I think the points are fair and valid, I will continue participant  based on my independent review of the applications.

**Comment by cryptowhizzard** [comment]
Points fair an valid. I will participate

**Comment by jamerduhgamer** [comment]
Hello @kevzak, we would also be interested in continuing to participate in the E-FIL+ application review for the year of 2023 🙂

**Reply by PluskitOfficial** [reply]
@kevzak Willing to participate

**Comment by MatrixStorage** [comment]
We are willing to participate.

**Comment by NiwanDao** [comment]
@kevzak I am interested in v1 E-FIl+

**Comment by Holiday507** [comment]
Pay attention and look forward to participating

**Reply by kevzak** [reply]
@Holiday507 can you send your notary application link? I cannot find you. Also send your slack handle. Thanks

**Comment by Suyanj** [comment]
We would like to participate and look forward to further progress.

**Comment by BlockMakeronline** [comment]
Really support! We'd like to participate.

**Comment by 1475Notary** [comment]
Happy to see the development of Efil+. We'll stay active and provide support!

**Comment by MetaWaveInfo** [comment]
Willing to participate. Thanks!

**Reply by kevzak** [reply]
@MetaWaveInfo can you send your notary application link? I cannot find you. Also send your slack handle. Thanks

**Comment by BobbyChoii** [comment]
Hi @kevzak, this's bobby from Round 4. I would like to join E-Fil notary group as well. Thank you.

**Comment by FILaway** [comment]
Solid points. I'm interested and willing to join.

**Comment by AthSmith** [comment]
Pls add me to the list, thank you!

**Comment by MEIYAN666** [comment]
willing to participate. thanks

**Comment by kevzak** [comment]
OK, thank you everyone for the comments. We are closing signups now to process the multisig. I will send a note on next steps soon.

**Reply by MRJAVAZHAO** [reply]
@kevzak  Willing to participate

**Comment by XnMatrixSV** [comment]
hi, this is Xnmatrix and I have commented on the slack channel willing to participate before 24th Jan! May I know if I can still participate as I didn't comment on GitHub?

**Comment by kevzak** [comment]
Discussion is closed, mulitsig addresses updated in https://github.com/filecoin-project/notary-governance/issues/613

**Comment by kevzak** [comment]
This is a reminder to all E-Fil+ Notaries to only sign applications where you have been tagged in the application comment by myself. If you have any questions, please ask. Thank you.

**Comment by kevzak** [comment]
An update for: 
@carohere
@joss-hua
@ewesion
@holiday507
@Aifabot-Cloud
@RawTechVentures
@TakiChain
@Meibuy
@Protocan
@OlivierMolenkamp
@BobbyChoii
@PangodGroup
@XnMatrixSV

you have been added to the E-Fil+ multisig. I will begin randomly assigning you to allocation requests as they come in. I recommend to review the [SLA details above](https://github.com/filecoin-project/notary-governance/discussions/807#discussion-4762523). Let me know if you have any questions.

**Comment by XnMatrixSV** [comment]
@kevzak Since we are the round 4 notary, does it mean that we can only start signing after we get out datacap? As out datacap now is still zero now.

**Comment by kevzak** [comment]
I don't understand what you are talking about here @XnMatrixSV . 

E-Fil is a separate multisig. Once assigned by me to review an application, you should have access to sign.

For all other, non-private, data application, you should have access as a v4 notary. If you do not, contact @Kevin-FF-USA

**Reply by XnMatrixSV** [reply]
okay understand, thank you!

**Comment by kevzak** [comment]
**Here an overview of the results thus far of v1 of the E-Fil+ random notary assignment pilot:**

- 20-35 notaries in the E-Fil Multisig (just added 15 new notaries in March 2023)
- 14 applications reviewed

- 106 notary random assignments 
  - 63 (59%) engaged (i.e. commented within the set 72 hr timeframe)
  - 43 (41%) did not respond at all 

- 46 of 63 (73%) signed the allocation request
  - 14 signed unprompted (were not assigned)
- 17 (27%) engaged but did not sign
   - 5 declined to sign (did not support)
   - 2 had a conflict of interest
   - 7 had issues signing (tooling/bot) and were reassigned
   - 3 never responded after initial engagement 

**Overall Takeaways:**
**1) Technically only 32 of 106 (30%) signed per guidelines**
- That's a lot of different notary time and energy to get to 32 signatures 
- Not to mention also the time the client has to invest answering questions, waiting for approvals and reassignments

**2) Engagement overall was high (73%) but just not on-time as needed**
- Accountability expectations need to be re-evaluated. This is a voluntary role and notaries likely had other priorities that limited their abilities to respond on-time, all the time. 
- If we set removal to three missed assignments, we will run out of notaries to assign

**3) Only 5% overall did not support an allocation**
- This speaks strongly in support of the upfront KYC/KYB/SP requirements as a method to curb potential abuse attempts
   
**4) After the first few allocation reviews, the comment thread gets to be too complicated to follow all the details** 
- Especially as we are randomly assigning completely new notaries each cycle, it takes time understand what is going on
- Need to find a way to ensure important details are labeled or pushed to the top of the application for quick access
- In general, each application needs to work through:
    - feedback from notaries about storage plan (SPs)
    - issues from CID Checker output
    - signing issues
    - bot messages
    - client updates to SPs, or plan to get back on track after CID failure

**5) No applications were closed due to consensus lack of support from notaries**
- However, in the event there is a situation where notaries look to close an application because of lack of support, what are guidelines to follow? For example, if 2 notaries are assigned and decline to sign and then an additional two also decline, this signals lack of support and no additional notaries would be assigned.

**Comment by kevzak** [comment]
Idea: Group of selected notaries work together with a client/lead SP throughout the duration of an application

Notaries would need to apply and meet minimum qualifications (TBD)

Phase 1 - Before 1st allocation: 
- Upfront trust building is vital to an application success
- Potential to meet client/SP virtually

Phase 2 - After 1st allocation: 
- The due diligence required moves toward CID Tracker checks
- Did you do what you said you were going do?

By assigning a group of notaries to an application from start to finish, we can focus efforts on:
- Building the very important upfront trust element with clients (preventing threats of abuse)
- Allowing notaries to be continuously aware and engaged with the of the progress of an application 
- Reducing the time and effort required by all parties involved in supporting, real clients w real data

**Comment by donkabat** [comment]
@kevzak 
Could you tell me, are you a volunteer or do you get paid? Why so important part of fil+ or efil+ is based on people who work totally for free? :(

**Reply by kevzak** [reply]
In regards to payment, there have been many previous discussions regarding notary incentives. Right now we're focused on social incentives.

Do you have any financial incentive ideas for notaries?

**Comment by donkabat** [comment]
If efil+ will work like it works now... maybe you can just kill it.
Notaries have too long time for reply. 1 bd for reaction and 2 bd for make a decision/take the needed action. 
We reply every time in few hours so why notaries have 2, 3, 4 days and after that you can only make next random assignment. It's ridiculous :(

**Comment by kevzak** [comment]
@donkabat - in the current Fil+ LDN system, notaries are free to sign any allocation, or not, as they wish. Within E-Fil we are testing engagement toward assignments of applications. This is a first step toward accountability and also looks to prevent collusion attempts for private datasets. See details here: https://github.com/filecoin-project/notary-governance/discussions/810

**Reply by donkabat** [reply]
yes, I know but if FIL+ and maybe EFIL+ is so important for PL you should just hire an employee as notary.

**Reply by donkabat** [reply]
maybe... employee(s) notary decide about asignments in 24h and after it 2 random notaries from community have next 24h to confirm or reject it but the decision of both community notaries must be the same to change first decision. If at least one of them does not make a decision within 24 hours, or if they are not the same, the first decision comes into force.
